lib/puppet/provider/java_ks/keytool.rb

```diff
@@ -26,50 +26,48 @@ def to_pkcs12(path)
       chain_certs = get_chain(certificate)
       x509_cert = chain_certs.shift
     end
-    pkcs12 = OpenSSL::PKCS12.create(get_password, @resource[:name], pkey, x509_cert, chain_certs)
-    File.open(path, "wb") { |f| f.print pkcs12.to_der }
+    pkcs12 = OpenSSL::PKCS12.create(password?, @resource[:name], pkey, x509_cert, chain_certs)
+    File.open(path, 'wb') { |f| f.print pkcs12.to_der }
   end
 
   # Keytool can only import a jceks keystore if the format is der.  Generating and
   # importing a keystore is used to add private_key and certifcate pairs.
   def to_der(path)
     x509_cert = OpenSSL::X509::Certificate.new File.read certificate
-    File.open(path, "wb") { |f| f.print x509_cert.to_der }
+    File.open(path, 'wb') { |f| f.print x509_cert.to_der }
   end
 
   def get_chain(path)
-    File.read(path).scan(/-----BEGIN [^\n]*CERTIFICATE.*?-----END [^\n]*CERTIFICATE-----/m).map {|cert| OpenSSL::X509::Certificate.new cert}
+    File.read(path).scan(%r{-----BEGIN [^\n]*CERTIFICATE.*?-----END [^\n]*CERTIFICATE-----}m).map { |cert| OpenSSL::X509::Certificate.new cert }
   end
 
-  def get_password
+  def password?
```

Comment:
typically a question mark denotes a boolean method but this returns a string. maybe nix the question mark.

Reply:
changed